### PR TITLE
Podman with Dev Services and Testcontainers

### PR DESCRIPTION
**Pull request by mjurc**
* Adding a blog post Podman configuration for Quarkus Dev Services and
  Testcontainers.

### CODE REVIEW

**Review by xstefank — COMMENTED** [review]
Maybe it would be good to also add TL;DR section to the beginning?

File: _posts/2021-11-02-quarkus-devservices-testcontainers-podman.adoc [hunk 1]
```diff
@@ -0,0 +1,114 @@
+---
+layout: post
+title: 'Configuring Podman for Quarkus Dev Services and Testcontainers'
+date: 2021-11-02
+tags: development-tips podman testcontainers
+synopsis: 'Quarkus Dev Services bring '
```
[the remark this answers is not in the document]
Comment (mjurc) [reply _posts/2021-11-02-quarkus-devservices-testcontainers-podman.adoc]: I've added a synopsis - can you check if this is sufficient?

File: _posts/2021-11-02-quarkus-devservices-testcontainers-podman.adoc [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+For our scenario, we will run a rootless Podman service. We need to set our environment up so that the clients that
+previously accessed the Docker Unix socket will know to reach the Podman service.
+
+By convention, the Docker clients attempt to connect to the service specified by URL configured in the `DOCKER_HOST`
+environment variable, so this variable needs to be configured to point to the Unix socket that the Podman service will
+be listening on:
+
+[source,bash]
+export DOCKER_HOST=unix:///run/user/${UID}/podman/podman.sock
```
[the remark this answers is not in the document]
Comment (mjurc) [reply _posts/2021-11-02-quarkus-devservices-testcontainers-podman.adoc]: I am actually not sure - RHEL docs mention URL with 3 slashes though:

https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/building_running_and_managing_containers/assembly_using-the-container-tools-api_using-the-container-tools-cli#proc_enabling-the-podman-api-using-systemd-in-rootless-mode_assembly_using-the-container-tools-api

File: _posts/2021-11-02-quarkus-devservices-testcontainers-podman.adoc [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+export TESTCONTAINERS_RYUK_DISABLED=true
+
+[NOTE]
+====
+This setting will only apply to the current terminal session. To make this configuration persistent, add the line to
+the profile files of your shell (e.g. `~/.profile`).
+====
+
+Podman is now available to respond to the Java Docker client used in Testcontainers. Please note that the Quarkus Dev
+Services require that a `docker` command is available on `PATH`. `podman-docker` package on Linux distributions provides
```
[the remark this answers is not in the document]
Comment (mjurc) [reply _posts/2021-11-02-quarkus-devservices-testcontainers-podman.adoc]: Added!

File: _posts/2021-11-02-quarkus-devservices-testcontainers-podman.adoc [hunk 4]
```diff
@@ -0,0 +1,114 @@
+---
+layout: post
+title: 'Configuring Podman for Quarkus Dev Services and Testcontainers'
```
[the remark this answers is not in the document]
Comment (mjurc) [reply _posts/2021-11-02-quarkus-devservices-testcontainers-podman.adoc]: Updated.

File: _posts/2021-11-02-quarkus-devservices-testcontainers-podman.adoc [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+
+* Running on a Linux system with Podman 3.x installed.
+* `podman-docker` installed to emulate the Docker CLI for Quarkus Dev Services
+* (optional) `podman-remote` installed for verification steps.
+
+== Configuration
+
+=== Migrating from Docker
+
+If you have previously been running Docker on a version that did not support CGroups V2 on modern Linux distributions,
```
[the remark this answers is not in the document]
Comment (mjurc) [reply _posts/2021-11-02-quarkus-devservices-testcontainers-podman.adoc]: I've moved the paragraph towards the end of article and added a command to check if the kernel argument is present - is this better?

File: _posts/2021-11-02-quarkus-devservices-testcontainers-podman.adoc [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+
+[source,bash]
+sudo grubby --update-kernel=ALL --remove-args="systemd.unified_cgroup_hierarchy=0"
+
+This setting will only take effect after a reboot.
+
+=== Configuring the Podman service
+
+Podman is daemonless container engine. Quarkus Dev Services and Testcontainers  expect a running Docker daemon listening 
+at a Unix socket. Before version 3, Podman was not able to listen on a Unix socket and only
```
[the remark this answers is not in the document]
Comment (mjurc) [reply _posts/2021-11-02-quarkus-devservices-testcontainers-podman.adoc]: I've removed the information.

File: _posts/2021-11-02-quarkus-devservices-testcontainers-podman.adoc [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+before changing this setting.
+====
+
+Finally, you can start the Podman service listening on the socket previously specified by the `DOCKER_HOST` environment
+variable. The following command will start the Podman service process in the current terminal session. Note that the
+Podman service launched by this command is not a daemon and will only survive for as long as it is open in the terminal.
+The command is blocking and the process does not output anything until a request is made against `$\{DOCKER_HOST\}`.
+
+[source,bash]
+podman system service --time=0 ${DOCKER_HOST}
```
[the remark this answers is not in the document]
Comment (mjurc) [reply _posts/2021-11-02-quarkus-devservices-testcontainers-podman.adoc]: Done - does this look better?

File: _posts/2021-11-02-quarkus-devservices-testcontainers-podman.adoc [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+This setting will only take effect after a reboot.
+
+=== Configuring the Podman service
+
+Podman is daemonless container engine. Quarkus Dev Services and Testcontainers  expect a running Docker daemon listening 
+at a Unix socket. Before version 3, Podman was not able to listen on a Unix socket and only
+responded as a rest service. Since version 3, Podman can be configured to create a service listening at a Unix socket
+and this service can be used with Dev Services and Testcontainers.
+
+For our scenario, we will run a rootless Podman service. We need to set our environment up so that the clients that
```
[the remark this answers is not in the document]
Comment (mjurc) [reply _posts/2021-11-02-quarkus-devservices-testcontainers-podman.adoc]: Right, podman-docker links `/var/lib/docker.sock` to `/run/podman/podman.sock` on F35 but... I found rootful podman a bit hard to work with as I have never figured out a simple way to get access to the podman socket from user account.

The docs show how to connect to rootful podman.sock from superuser. But I guess most users don't run stuff as superuser for development.

Is there anything I've missed? 

I've also added some information about what rootless podman means.

**Review by xstefank — APPROVED** [review]
LGTM but I still think a short TL;DR section, in the beginning, can be beneficial as not everyone needs/wants to go into details.

[source,bash]
----
# install podman and podman-docker packages
systemctl --user enable podman.socket --now
export DOCKER_HOST=unix:///run/user/${UID}/podman/podman.sock
export TESTCONTAINERS_RYUK_DISABLED=true
----

but if that's only my opinion feel free to discard it.

**Comment by gsmet** [comment]
@mjurc can you check @xstefank 's comment above? Once you give me the green light, I'll schedule it for publishing sometime next week.

**Review by gsmet — APPROVED** [review]

**Comment by mjurc** [comment]
@gsmet @xstefank I've added the short TL;DR section. Feel free to proceed with publish.

**Review by xstefank — APPROVED** [review]
perfect, thanks @mjurc

**Pull request merged**